src/ImageCapture_Gst.cpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <assert.h>
+#include <gst/gst.h>
+#include <sstream>
+#include <unistd.h>
+#include <vector>
+
+#include "CameraParameters.h"
+#include "ImageCapture_Gst.h"
+
+#include "log.h"
+
+ImageCapture_Gst::ImageCapture_Gst(std::shared_ptr<CameraDevice> camDev)
+    : mCamDev(camDev)
+    , mState(STATE_IDLE)
+    , mWidth(640)
+    , mHeight(480)
+    , mImgFormat(CameraParameters::ID_IMAGE_FORMAT_JPEG)
+    , mPixelFormat(-1)
+    , mImgPath("/tmp/")
+    , mResultCB(nullptr)
+{
+}
+
+ImageCapture_Gst::~ImageCapture_Gst()
+{
+    stop();
+}
+
+int ImageCapture_Gst::start(int interval, int count,
+                            std::function<void(int result, int seq_num)> cb)
+{
+    log_debug("%s interval:%d count:%d", __func__, interval, count);
+    // Invalid Arguments
+    // Either the capture is count based or interval based or count with interval
+    if (count <= 0 && interval <= 0) {
+        log_error("Invalid Parameters");
+        return 1;
+    }
+
+    // check & set state
+    if (mState != STATE_IDLE)
+        return 1;
+
+    mResultCB = cb;
+    setState(STATE_IN_PROGRESS);
+    // create a thread to capture images
+    mThread = std::thread(&ImageCapture_Gst::capture_thread, this, count, interval);
+
+    return 0;
+}
+
+int ImageCapture_Gst::stop()
+{
+    if (mState == STATE_IDLE)
+        return 0;
+
+    setState(STATE_IDLE);
+
+    // TODO::thread_join if mThread legal
+    return 0;
+}
+
+int ImageCapture_Gst::setState(int state)
+{
+    mState = state;
+
+    return 0;
+}
+
+int ImageCapture_Gst::getState()
+{
+    return mState;
+}
+
+int ImageCapture_Gst::setResolution(int imgWidth, int imgHeight)
+{
+    mWidth = imgWidth;
+    mHeight = imgHeight;
+
+    return 0;
+}
+
+int ImageCapture_Gst::setFormat(int imgFormat)
+{
+    mImgFormat = imgFormat;
+
+    return 0;
+}
+
+int ImageCapture_Gst::setDestination(const std::string imgPath)
+{
+    // TODO::Check if the path is writeable
+    mImgPath = imgPath;
+
+    return 0;
+}
+
+void ImageCapture_Gst::capture_thread(int num, int interval)
```

Comment:
Method is in snake_case but others are in CamelCase; I suggest `ImageCaptureGst::start_capture_thread()`.

Reply:
Fixed